components/classroom/main.js

```diff
@@ -42,3 +45,50 @@ window.onload = () => {
         console.log('Project ID not found in URL');
     }
 };
+
+const ROLE_HIERARCHY = [Roles.OWNER, Roles.LEADER, Roles.CONTRIBUTOR];
+
+// identifies the highest privilege role from a list of user roles and returns the top-ranked valid role
+function getHighestPrivilegeRole(userRoles) {
+    const validRoles = [];
+    const invalidRoles = [];
+
+    // Loop through each role in userRoles
+    for (let i = 0; i < userRoles.length; i++) {
+        const role = userRoles[i];
+
+        // Check if the role is in ROLE_HIERARCHY
+        if (ROLE_HIERARCHY.includes(role)) {
+            validRoles.push(role); // Add to validRoles if it's a recognized role
+        } else {
+            invalidRoles.push(role); // Add to invalidRoles if it's unrecognized
+        }
+    }
+
+    // Log unexpected roles if any were found
+    if (invalidRoles.length > 0) {
```

Comment:
Not sure there's a need for an invalidRoles array? If role is not within hierarchy just throw error